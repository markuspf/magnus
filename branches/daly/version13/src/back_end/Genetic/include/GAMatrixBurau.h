// Copyright (C) 1999 The New York Group Theory Cooperative
// See magnus/doc/COPYRIGHT for the full notice.
//
// Contents: Definition of class GAMatrixBurau
//
// Principal Author: Dmitry Bormotov
//
// Status: in progress
//
// Description:
//
// Revision History:
//


#ifndef _GAMatrixBurau_H_
#define _GAMatrixBurau_H_


#include "PMDebornoyWord.h"
#include "PMArray.h"
#include "GA.h"
#include "FreeGroup.h"
#include "Matrix.h"
#include "Polynomial.h"
#include "Int2.h"


// --------------------------- GAMatrixBurau ------------------------------- //


class GAMatrixBurau : public GA
{

public:
  
  /////////////////////////////////////////////////////////////////////////
  //                                                                     //
  // Constructors:                                                       //
  //                                                                     //
  /////////////////////////////////////////////////////////////////////////
  
  GAMatrixBurau( const GAConfig& GAC, const FreeGroup& Br );
  
  ~GAMatrixBurau( );
  
  // copy constructor and operator '=' are hidden (not implemented)


  /////////////////////////////////////////////////////////////////////////
  //                                                                     //
  // Overriden abstract functions:                                       //
  //                                                                     //
  /////////////////////////////////////////////////////////////////////////

  void initPopulation( );

  int fitness( const PM* pm );
  

  /////////////////////////////////////////////////////////////////////////
  //                                                                     //
  // Accessors:                                                          //
  //                                                                     //
  /////////////////////////////////////////////////////////////////////////

  Word getSolution( ostream* o = NULL );


  /////////////////////////////////////////////////////////////////////////
  //                                                                     //
  // OI:                                                                 //
  //                                                                     //
  /////////////////////////////////////////////////////////////////////////


private:

  /////////////////////////////////////////////////////////////////////////
  //                                                                     //
  // Private functions:                                                  //
  //                                                                     //
  /////////////////////////////////////////////////////////////////////////

  // copy constructor and operator '=' are hidden (not implemented)
  
  GAMatrixBurau( const GAMatrixBurau& );

  GAMatrixBurau& operator = ( const GAMatrixBurau& );
  
  void init( );
    
  bool checkForSolution( Word& res );

  Matrix< Polynomial<Int2> > productMatrix( const PM* pm );

  void penalizeFitness( );


  /////////////////////////////////////////////////////////////////////////
  //                                                                     //
  // Data members:                                                       //
  //                                                                     //
  /////////////////////////////////////////////////////////////////////////
  
  FreeGroup B;
  int numOfGens, n, g;
  bool keepDetails;
  ostream* out;
  Matrix< Polynomial<Int2> > *bgens, *bgensI, I;
  PMDebornoyWordConfig PMDWC;
};


#endif
