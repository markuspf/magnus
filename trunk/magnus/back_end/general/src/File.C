#include "File.h"
#include <iostream>

#define BUFSIZE 1000
void dumpstream(istream& src, ostream& trg) {
	char buf[BUFSIZE];
	unsigned long sz(1);
	while(src.good() && (sz > 0)) {
		sz = src.readsome(buf, BUFSIZE);
		trg.write(buf, sz);
	}
}
