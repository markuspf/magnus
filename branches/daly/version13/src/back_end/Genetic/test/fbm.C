// Copyright (C) 1997 The New York Group Theory Cooperative
// See magnus/doc/COPYRIGHT for the full notice.

// Contents: Informal test.
//
// Principal Author: Dmitry Bormotov
//
// Status:
//
// Revision History:
//


#include <values.h>
#include "FreeGroup.h"
#include "Config.h"
#include "RandomNumbers.h"
#include "GAMatrixBurau.h"
#include "GAFastBurMat.h"
#include "File.h"


int main ( )
{
  Map M;
  Chars errMsg;
  GAConfig c(50,-1,100,100,GAConfig::STRONG,true); 
  cout << c << endl;
 
  FreeGroup F;
  cout << "Enter a free group: ";
  errMsg = cin >> F;
  if( errMsg.length() > 0 ) {
    cout << errMsg;
    exit(0);
  }
  cout << endl;

  GAMatrixBurau solverT(c, F); 
  GAFastBurMat solver(c, F); 


  Word solution = solver.getSolution(&cout);
  cout << "The solution is ";
  F.printWord(cout,solution);
  cout << endl;

  /*
  BurauPoly p(1),q(p);
  cout << "p = " << p << endl;
  p.multByIX();
  cout << "p*x^-1 = " << p << endl;
  q.sub(p);
  cout << "p*(1-x^-1) = " << q << endl;
  */
}

