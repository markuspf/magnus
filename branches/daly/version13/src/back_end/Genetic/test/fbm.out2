Population: 
0: 
[ -5 -4 1 4 -5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 -5 -4 -2 5 ]
1: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
2: 
[ 4 1 -5 1 5 1 5 4 -2 -5 -4 2 1 1 -5 -4 1 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
3: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 1 1 4 3 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 ]
4: 
[ 1 1 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
5: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 4 -5 -4 -4 1 4 -5 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
6: 
[ 1 1 -5 1 5 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
7: 
[ 4 1 -5 1 5 1 5 4 -2 -5 -4 2 1 1 -5 -4 1 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
8: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 1 4 -2 -3 5 3 -2 1 4 5 2 -5 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
9: 
[ 1 1 -5 1 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
10: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
11: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
12: 
[ 1 1 1 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
13: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 ]
14: 
[ 1 1 -5 1 1 5 1 5 1 -2 -5 -4 2 1 4 -5 -4 -4 1 4 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
15: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 5 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
16: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 1 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
17: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
18: 
[ 4 1 -5 1 5 1 5 4 -2 -5 -4 2 1 1 -5 -4 1 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
19: 
[ 1 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
20: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 4 5 -4 2 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
21: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -4 2 1 1 4 -5 -4 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
22: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
23: 
[ 1 1 1 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
24: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 -5 -4 -4 1 4 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 -5 -4 -2 5 ]
25: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -5 -4 1 4 -2 -3 5 -4 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
26: 
[ -5 -4 1 4 -5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 5 4 5 2 4 -5 -4 -4 1 4 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 2 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
27: 
[ 1 1 -5 1 1 5 1 5 1 -2 -5 -4 2 1 4 -5 -4 -4 1 4 4 2 -3 5 3 -2 1 4 5 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
28: 
[ 1 -5 1 -2 -3 5 3 -2 1 4 5 2 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
29: 
[ -3 1 3 -2 -3 1 3 -2 -5 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 -2 -2 -2 1 4 5 2 1 1 4 1 3 -2 -3 1 3 -4 1 -5 -4 -2 5 ]
30: 
[ 1 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
31: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 1 1 4 -5 -4 1 5 -2 -3 5 3 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 4 5 -4 2 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
32: 
[ 1 4 1 -5 1 5 1 5 1 -2 -5 -4 2 -3 1 -4 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
33: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 -5 -4 -2 ]
34: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 4 -5 -4 1 4 -2 -3 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
35: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 5 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -5 -4 1 4 4 1 4 5 2 4 -5 -4 -4 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 1 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
36: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 5 1 4 5 2 4 -5 -4 3 -2 -2 1 4 5 2 1 4 4 -5 -4 -2 ]
37: 
[ 1 1 1 -5 -4 1 4 -2 -3 5 3 -2 3 -2 -2 1 4 -5 -4 -4 1 4 -5 -3 1 5 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
38: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 1 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 1 4 1 3 -2 -3 1 3 -4 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
39: 
[ 1 1 2 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 1 4 -5 -4 -4 2 1 1 ]
40: 
[ 1 1 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
41: 
[ 1 1 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 1 5 1 -2 -5 -4 2 1 -4 -5 4 5 1 4 -5 -4 1 4 -5 -4 1 4 -2 -3 5 3 -2 -3 4 5 2 4 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 4 -5 -4 -2 ]
42: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 5 3 -2 5 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -3 1 3 -2 -2 1 4 5 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -4 -2 ]
43: 
[ 4 1 -5 1 5 1 5 1 1 1 4 -5 -4 -2 ]
44: 
[ 1 1 -5 1 5 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 4 -5 -4 1 4 -2 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -4 2 1 1 4 -5 -4 1 4 4 1 4 5 2 4 -5 -4 -4 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 1 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
45: 
[ 1 1 -5 1 1 5 1 5 1 -2 -5 -4 2 1 4 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 5 1 -2 -5 -4 2 1 1 4 -5 -4 1 3 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -4 -2 ]
46: 
[ 1 1 1 -5 -4 1 4 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 1 1 4 -5 -5 -2 ]
47: 
[ 1 1 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 1 1 4 -5 -4 -2 ]
48: 
[ 4 1 -5 1 5 1 5 1 -2 -5 -4 2 1 -4 1 4 5 -4 1 1 1 4 -5 -4 1 4 -2 -3 5 3 -4 2 1 1 4 -5 -4 -5 1 -2 -3 5 3 -2 1 4 5 2 4 -5 -4 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -4 1 4 -5 -3 1 3 -2 -3 1 3 -2 -2 1 4 5 2 -5 -4 -4 1 4 -5 1 4 -5 -4 -2 ]
49: 
[ 4 1 -5 1 5 1 -5 1 -2 -5 -4 2 1 -4 -5 4 -2 ]
Generation: 364500   Penalized fitness: 2
Penalized fitness values: 
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 11 6 9 7 19 12 16 26 6 12 15 12 26 20 15 11 20 20 36 12 16 10 12 8 37
