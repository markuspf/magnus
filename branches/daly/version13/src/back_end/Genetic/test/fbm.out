

1. Matrix: 

|  1  - x1 + x1^2  - x1^3 + x1^4  - x1^5 + x1^6  - x1^7 + x1^8 ,  
x1  - x1^2 + x1^3  - x1^4 + x1^5  - x1^6 + x1^7  - x1^8 , 
0, 0, 0, 0, |

|  1  - x1 + x1^2  - x1^3 + x1^4  - x1^5 + x1^6  - x1^7 ,  
x1  - x1^2 + x1^3  - x1^4 + x1^5  - x1^6 + x1^7 
, 0, 0, 0, 0, |

| 0, 0,  1 , 0, 0, 0, |
| 0, 0, 0,  1 , 0, 0, |
| 0, 0, 0, 0,  1 , 0, |
| 0, 0, 0, 0, 0,  1 , |.

Fitness = 2

The solution is 

g5^-1 g4^-1 g1 g4 g5^-1 g4^-1 g1^3 g4 g5^-1 g4^-1 g1 g4 g2^-1 g3^-1 g5
g3 g2^-1 g1 g4 g5 g2 g4 g5^-1 g4^-2 g1 g4 g3^-1 g1 g3 g2^-1 g3^-1 g1
g3 g2^-2 g1 g4 g5 g2 g1^2 g5^-1 g4^-1 g2^-1 g5


2. Matrix: 

|  1  - x1 + x1^2  - x1^3 + x1^4  - x1^5 + x1^6  - x1^7 + x1^8  - x1^9 
+ x1^10  - x1^11 + x1^12 ,  
x1  - x1^2 + x1^3  - x1^4 + x1^5  - x1^6 + x1^7  - x1^8 + x1^9  - x1^10 
+ x1^11  - x1^12 
, 0, 0, 0, 0, |

|  1  - x1 + x1^2  - x1^3 + x1^4  - x1^5 + x1^6  - x1^7 + x1^8  - x1^9 
+ x1^10  - x1^11 ,  
x1  - x1^2 + x1^3  - x1^4 + x1^5  - x1^6 + x1^7  - x1^8 + x1^9  - x1^10 
+ x1^11 
, 0, 0, 0, 0, |

| 0, 0,  1 , 0, 0, 0, |
| 0, 0, 0,  1 , 0, 0, |
| 0, 0, 0, 0,  1 , 0, |
| 0, 0, 0, 0, 0,  1 , |.

Fitness = 2

The solution is 

g1^2 g5^-1 g1 g5 g1 g5 g1 g2^-1 g5^-1 g4^-1 g2 g1^2 g4 g5^-1 g4^-1 g1
g4 g5^-1 g4^-1 g1 g4 g2^-1 g3^-1 g5 g3 g2^-1 g1 g4 g5 g2 g4 g5^-1
g4^-2 g1 g4 g5^-1 g3^-1 g1 g3 g2^-1 g3^-1 g1 g3 g2^-2 g1 g4 g5 g2 g1^2
g4 g5^-1 g4^-1 g2^-1


3. Matrix: 

|  1  - x1 + x1^2  - x1^3 + x1^4  - x1^5 + x1^6  - x1^7 + x1^8  - x1^9 
+ x1^10  - x1^11 + x1^12  - x1^13 + x1^14  - x1^15 + x1^16  - x1^17 + x1^18 ,
x1  - x1^2 + x1^3  - x1^4 + x1^5  - x1^6 + x1^7  - x1^8 + x1^9  - x1^10 
+ x1^11  - x1^12 + x1^13  - x1^14 + x1^15  - x1^16 + x1^17  - x1^18 
, 0, 0, 0, 0, |

|  1  - x1 + x1^2  - x1^3 + x1^4  - x1^5 + x1^6  - x1^7 + x1^8  - x1^9 + 
x1^10  - x1^11 + x1^12  - x1^13 + x1^14  - x1^15 + x1^16  - x1^17 ,
x1  - x1^2 + x1^3  - x1^4 + x1^5  - x1^6 + x1^7  - x1^8 + x1^9  - x1^10 
+ x1^11  - x1^12 + x1^13  - x1^14 + x1^15  - x1^16 + x1^17 
, 0, 0, 0, 0, |

| 0, 0,  1 , 0, 0, 0, |
| 0, 0, 0,  1 , 0, 0, |
| 0, 0, 0, 0,  1 , 0, |
| 0, 0, 0, 0, 0,  1 , |.

Fitness = 2

The solution is 

g4 g1 g5^-1 g1 g5 g1 g5 g4 g2^-1 g5^-1 g4^-1 g2 g1^2 g5^-1 g4^-1 g1^2
g4 g2^-1 g3^-1 g5 g3 g2^-1 g1 g4 g5 g2 g4 g5^-1 g4^-2 g1 g4 g5^-1
g3^-1 g1 g3 g2^-1 g5 g1 g2^-1 g5^-1 g4^-1 g2 g1^2 g4 g5^-1 g4^-1 g1 g4
g5^-1 g4^-1 g1 g4 g2^-1 g3^-1 g5 g3 g2^-1 g1 g4 g5 g2 g4 g5^-1 g4^-2
g1 g4 g5^-1 g3^-1 g1 g3 g2^-1 g3^-1 g1 g3 g2^-2 g1 g4 g5 g2 g5^-1
g4^-2 g1 g4 g5^-1 g3^-1 g1 g3 g2^-1 g3^-1 g1 g3 g2^-2 g1 g4 g5 g2 g1^2
g4 g5^-1 g4^-1 g2^-1





